src/Microsoft.DotNet.Interactive.HttpRequest.Tests/ParserTests.cs

```diff
@@ -0,0 +1,595 @@
+                <request>
+                    <name>sample</name>
+                    <time>Wed, 21 Oct 2015 18:27:50 GMT</time>
+                </request>
+                """);
+        }
+    }
+
+    public class Comment
+    {
+        [Fact]
+        public void comments_are_parsed_correctly()
+        {
+
+            var result = HttpRequestParser.Parse(
+                """
+                # This is a comment
+                GET https://example.com HTTP/1.1"
+                """);
+
+            using var _ = new AssertionScope();
+
+            var methodNode = result.SyntaxTree.RootNode
+                  .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which
+                  .MethodNode;
+
+            methodNode.ChildNodes.Should().ContainSingle<HttpCommentNode>().Which.Text.Should().Be(
+              "# This is a comment");
+
+
+        }
+    }
+}
+// TODO: Test string with variable declarations but no requests
+
+/*
+[TestClass]
+public class TokenTest
+{
+
+    [Fact]
+    public async Task RequestWithEmbeddedDynamicVariableWithSpacesAsync()
+    {
+        string text = $"\r{Environment.NewLine}GET https://example.com?id={{{{$randomInt 1 2}}}}&id2=value&id3={{{{$randomInt 2 3}}}} http/1.1";
+
+        IRestDocumentSnapshot doc = await TestHelpers.CreateDocumentSnapshotAsync(text);
+
+        Request request = doc.Requests[0];
+
+        Assert.IsNotNull(request.Version);
+        Assert.AreEqual("http/1.1", request.Version.Text);
+
+        Assert.AreEqual("https://example.com?id={{$randomInt 1 2}}&id2=value&id3={{$randomInt 2 3}}", request.Url.Text);
+    }
+
+    [Fact]
+    public async Task MultipleRequestsAsync()
+    {
+        string text = """
+            get http://example.com
+
+            ###
+
+            post http://bing.com
+            """;
+
+        IRestDocumentSnapshot doc = await TestHelpers.CreateDocumentSnapshotAsync(text);
+
+        Assert.AreEqual(2, doc.Requests.Count);
+    }
+
+    [Fact]
+    public async Task RequestWithHeaderAndBodyAndCommentAsync()
+    {
+        string text = """
+            DELETE https://example.com
+            User-Agent: ost
+            #ost:hat
+
+            {
+                "enabled": true
+            }
+
+            ###
+            """;
+
+        IRestDocumentSnapshot doc = await TestHelpers.CreateDocumentSnapshotAsync(text);
+        Request first = doc.Requests[0];
+
+        Assert.IsNotNull(first);
+        Assert.AreEqual(1, doc.Requests.Count);
+        Assert.AreEqual(ItemType.HeaderName, first.Children[2].Type);
+        Assert.AreEqual(ItemType.HeaderValue, first.Children[3].Type);
+        Assert.AreEqual(ItemType.Comment, first.Children[4].Type);
+        Assert.AreEqual(ItemType.EmptyLine, first.Children[5].Type);
+        Assert.AreEqual(ItemType.Body, first.Children[6].Type);
+        Assert.AreEqual("{\r\n    \"enabled\": true\r\n}\r\n", first.Body);
+    }
+
+    [Fact]
+    public async Task BodyAfterCommentAsync()
```

Comment:
This test should work now, should it not? The comment should be added as leading trivia for the body.